### ISSUE TITLE
twisted/web/http.py has self._header instead of self.__header

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2683]
|<img alt="jvtm's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jvtm reported|
|-|-|
|Trac ID|trac#2683|
|Type|defect|
|Created|2007-05-30 13:03:27Z|

I think [twisted/web/http.py@20346#L1062](twisted/web/http.py@20346#L1062) should clear ```self.__header```, not ```self._header```.

<details><summary>Searchable metadata</summary>

```
trac-id__2683 2683
type__defect defect
reporter__jvtm jvtm
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1180530207000000 1180530207000000
changetime__1221587646000000 1221587646000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Indeed, `_header` isn't anything.  However, clearly it's not needed to reset this attribute, since it has never been reset and things work fine.  Inspecting the code, this makes sense - `__header` is always reset to `''` by some other part of the code before `allContentReceived` is called, so it is redundant to set it there as well.

Here's my proposed fix:

```
Index: twisted/web/http.py
===================================================================
--- twisted/web/http.py (revision 24691)
+++ twisted/web/http.py (working copy)
@@ -1249,7 +1249,6 @@

         # reset ALL state variables, so we don't interfere with next request
         self.length = 0
-        self._header = ''
         self._receivedHeaderCount = 0
         self.__first_line = 1
         del self._command, self._path, self._version
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Looks sensible, please apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [24875]) Apply patch removing dead code from `HTTPChannel`

Author: exarkun
Reviewer: therve
Fixes: [#2683](https://github.com/twisted/twisted/issues/2683)

`HTTPChannel.allContentReceived` needlessly set a `_header` attribute
on itself, trying to reset the `__header` attribute.  The `__header`
attribute is reset elsewhere though, so just delete this broken code.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
